### DISCUSSION TITLE
Auction Draft Support?

### DISCUSSION THREAD

**Discussion by Ace7866** [mattdodge/yahoofantasy#17] [Ideas]
How difficult would it be to add an additional column to the CLI draft dump listing the $ amount paid for each draft pick for auction draft leagues?

**Comment by mattdodge** [comment]
Probably wouldn't be too hard - I haven't looked at the API response but if auction value is in there then it'd be pretty trivial to add it. Would probably be a pretty easy PR if you want to take a crack at it @Ace7866

**Reply by Ace7866** [reply]
Ooo exciting! That might be a great thing for me to take a stab at to help further my Git skills. As you'll see from my profile I'm a Git noob.

**Reply by Ace7866** [reply]
![image](https://user-images.githubusercontent.com/58792096/121238242-83949a80-c85d-11eb-82e0-58540015288f.png)

Hi Matt! Bear with me here.. this might be a silly question. When you refer to the API response, is this (screenshot above) where I could see if that information is contained? I'm making my way through your folders and the various files, and it seems like if I figure out how to print the returned value of this function draft_results(), this is where I'd see if the $ draft value is listed. Is that right? 

I logged in, imported Context() and then League and made an attempt to print(League.draft_results()) and got the missing 1 required positional argument [self] error. I only barely learned about instantiating classes at the end of my online course so I'm not super well-versed. Been doing some googling and reading though and confident I can eventually figure it out. But circling back... if I can figure out how to return the value from this function in your code, is there where I'd see if the $ auction bid is listed? From here I think I might know where/how to modify the CLI code to include the new information in the dump. 

Thanks again for your willingness to entertain a newcomer :-)

**Reply by Ace7866** [reply]
Ok, nevermind.. I now realize that will only reveal the results you've already structure. I need to go into the .api folder and look through there.

**Reply by mattdodge** [reply]
If the data is actually on those draft result objects from the API then it would be on the objects that come back in the list that is returned by `draft_results`. I'm not sure that the auction amount is on there though.

As you said, if you dig a bit more into the API you'll see that eventually the `_load_or_fetch` method is actually making an API request to the Yahoo Fantasy API. You'll be able to parse and dig through that raw data after the `_load_or_fetch` call.

The other option is to explore the [Yahoo API docs](https://developer.yahoo.com/fantasysports/guide/) or just play with the REST HTTP API directly. Getting those set up is a bit more involved, but if you're looking for new data from that API that is not currently available via this library it might be necessary I'm afraid.

Let me take a quick look through there and see if I see anything related to auction values. One of my baseball leagues is an auction league so I have an example to work off of.

**Comment by mattdodge** [comment]
@Ace7866 I think you're in luck! The auction amount appears to be on the draft result object under the `cost` attribute. Here's a snippet to get you started.

```python
c = Context()
lg = c.get_leagues('mlb', 2020)[0]
for result in lg.draft_results():
    print(result.player, result.cost)
```

If your league was not an auction league you'll see an AttributeError if you try to access cost.

**Reply by Ace7866** [reply]
Awesome!!! Thank you Matt! I'll fool around with this today/tomorrow.

**Reply by Ace7866** [reply]
![image](https://user-images.githubusercontent.com/58792096/122597972-c6aaf680-d031-11eb-9547-0f9dee7e9d29.png)
@mattdodge would the change be as simple was what's shown here? I was wondering if I would need to use a conditional to account for the AttributeError you described above for non-auction leagues? If result.cost == AttributeError, Pass... or something like that.

**Reply by mattdodge** [reply]
Yeah we should use a conditional here, or just put a `None` in there if there are no costs associated. Look up "handling exceptions" in python to see more about that, your syntax isn't quite right there.

I'm fine having the cost column always on the output, but only filled in in the case of an auction draft. So they can be blank for snake drafts. That should be relatively easy to program too using something like `getattr`.
